aws_analysis_tools/cli/update_ec2_tags.py

```diff
@@ -43,120 +41,132 @@ def add_cli_arguments(self, parser):
         )
 
         group.add_argument(
             '--environment',
-            default = 'dev',
-            help    = "The environment to set. (default: %(default)s)."
+            default='dev',
+            help="The environment to set. (default: %(default)s)."
         )
 
         group.add_argument(
             '--cluster-name',
-            required = True,
-            help     = "The cluster name to set. Example: 'apiservices-a'"
+            required=True,
+            help="The cluster name to set. Example: 'apiservices-a'"
         )
 
         group.add_argument(
             '--classes',
-            nargs   = "*",
-            default = [],
-            help    = "Specify class to tag with (can be used multiple times). "
-            "(default: %(default)s)"
+            nargs="*",
+            default=[],
+            help=("Specify class to tag with (can be used multiple times). "
+                  "(default: %(default)s)")
+        )
+
+        group.add_argument(
+            '--os',
+            default=None,
+            help=("The name of the OS to set. Example: Trusty"),
         )
 
         group.add_argument(
             '--dry-run',
-            default = False,
-            action  = 'store_true',
-            help    = "Do a dry run; find the tags, but do not update them "
-            "in EC2. (default: %(default)s)"
+            default=False,
+            action='store_true',
+            help=("Do a dry run; find the tags, but do not update them "
+                  "in EC2. (default: %(default)s)")
         )
 
-    def update_tags(self,
-        ec2_region      = None,
-        instance_id     = None,
-        classes         = None,
-        environment     = None,
-        cluster_name    = None,
-        dry_run         = False,
+    def update_tags(
+        self,
+        ec2_region=None,
+        instance_id=None,
+        classes=None,
+        environment=None,
+        cluster_name=None,
+        os=None,
+        dry_run=False,
     ):
-        log   = self.logger
+        log = self.logger
         stats = self.stats
 
-        ### from cli or passed explicitly?
-        ec2_region      = ec2_region    or self.ec2_region
-        instance_id     = instance_id   or self.instance_id
-        classes         = classes       or self.classes
-        dry_run         = dry_run       or self.dry_run
-        environment     = environment   or self.environment
-        cluster_name    = cluster_name  or self.cluster_name
-
+        # from cli or passed explicitly?
+        ec2_region = ec2_region if ec2_region is not None else self.ec2_region
+        instance_id = instance_id if instance_id is not None else self.instance_id
+        classes = classes if classes is not None else self.classes
+        dry_run = dry_run if dry_run is not None else self.dry_run
+        environment = environment if environment is not None else self.environment
+        os = os if os is not None else self.os
```

Comment:
@joestump This application actually is `exec`'d by puppet on each run. Puppet manifest change for this to only run when needed is on-going in my head.

We can figure out the OS from AMI and turn this into an optional arguments.